File: go/service/book_service.go
package service

import (
	"github.com/csandiego/poc-graphql/go/data"
)

type BookService interface {
	Create(data.Book) error
	Delete(int) error
	Get(int) (data.Book, error)
	List() ([]data.Book, error)
	Update(int, data.Book) error
}

File: ApolloJvm/src/main/kotlin/com/github/csandiego/pocgrpahql/apollojvm/service/BookService.kt
package com.github.csandiego.pocgrpahql.apollojvm.service

import com.github.csandiego.pocgrpahql.apollojvm.data.Book

interface BookService {

    suspend fun list(): List<Book>

    suspend fun get(id: Int): Book

    suspend fun create(book: Book)

    suspend fun update(id: Int, book: Book)

    suspend fun delete(id: Int)
}